test/python/test_dagcircuit.py

```diff
@@ -565,7 +565,7 @@ def test_remove_non_op_node(self):
         """Try to remove a non-op node with remove_op_node method."""
         self.dag.apply_operation_back(HGate(), [self.qubit0])
 
-        in_node = next(self.dag.topological_nodes())
+        in_node = next(iter(self.dag.topological_nodes()))
```

Comment:
I don't think the `iter` is needed anymore.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/3810/commits/af1905120381afdee3597bebe09769a1ab51d334